lib/ASRouterTriggerListeners.jsm

```diff
@@ -40,12 +41,24 @@ function isPrivateWindow(win) {
  *
  * @returns {string} - the host that matched the whitelist
  */
-function checkHost(aLocationURI, hosts, aRequest) {
+function checkHost(aLocationURI, {hosts, matchPatternSet}, aRequest) {
   // Check current location against whitelisted hosts
   if (hosts.has(aLocationURI.host)) {
     return aLocationURI.host;
   }
 
+  if (matchPatternSet) {
+    const uri = Services.io.newURI(aLocationURI.spec);
```

Comment:
I guess you can't use `aLocationURI` directly?

Reply:
Hm actually I should be able, I see matches has 2 implementations and one of them takes `strings`.